### ISSUE TITLE
UI test B43663 needs to be automated

### ISSUE THREAD

**Issue by samhouts** [xamarin/Xamarin.Forms#2374]
- [ ] Test needs to be automated
- [ ] Manual test needs instructions

**Comment by puppetSpace** [comment]
I have an issue running the UITest in UWP.
I installed the WinAppDriver and the test opens up the application. But when it is in the "Test Cases" view, the test times out and throws following exception:

```
[10/04/2019` 10:38:18 Informational] Both navigation methods failed. OpenQA.Selenium.WebDriverException: The HTTP request to the remote WebDriver server for URL http://127.0.0.1:4723/session/95208BD3-BC44-41B0-80EF-AE324A19452B/elements timed out after 60 seconds. ---> System.Net.WebException: The request was aborted: The operation has timed out.
   at System.Net.HttpWebRequest.GetResponse()
   at OpenQA.Selenium.Remote.HttpCommandExecutor.MakeHttpRequest(HttpRequestInfo requestInfo)
   --- End of inner exception stack trace ---
   at OpenQA.Selenium.Appium.Service.AppiumCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Appium.AppiumDriver`1.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteWebDriver.FindElements(String mechanism, String value)
   at OpenQA.Selenium.Appium.AppiumDriver`1.FindElements(String selector, String value)
   at OpenQA.Selenium.Appium.AppiumDriver`1.FindElementsByAccessibilityId(String selector)
   at Xamarin.Forms.Core.UITests.WinDriverApp.QueryWindows(WinQuery query) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 823
   at Xamarin.Forms.Core.UITests.WinDriverApp.QueryWindows(Func`2 query) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 849
   at Xamarin.Forms.Core.UITests.WinDriverApp.<>c__DisplayClass91_0.<WaitForElement>b__0() in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 551
   at Xamarin.Forms.Core.UITests.WinDriverApp.Wait(Func`1 query, Func`2 satisfactory, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 989
   at Xamarin.Forms.Core.UITests.WinDriverApp.WaitForAtLeastOne(Func`1 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 1012
   at Xamarin.Forms.Core.UITests.WinDriverApp.WaitForElement(Func`2 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency, Nullable`1 postTimeout) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 552
   at Xamarin.Forms.Controls.ScreenshotConditionalApp.WaitForElement(Func`2 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency, Nullable`1 postTimeout) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\TestPages\ScreenshotConditionalApp.cs:line 195
   at Xamarin.Forms.Controls.AppSetup.NavigateToIssue(Type type, IApp app) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\TestPages\TestPages.cs:line 182
[10/04/2019 10:38:18 Informational] Both navigation methods failed. OpenQA.Selenium.WebDriverException: The HTTP request to the remote WebDriver server for URL http://127.0.0.1:4723/session/95208BD3-BC44-41B0-80EF-AE324A19452B/elements timed out after 60 seconds. ---> System.Net.WebException: The request was aborted: The operation has timed out.
   at System.Net.HttpWebRequest.GetResponse()
   at OpenQA.Selenium.Remote.HttpCommandExecutor.MakeHttpRequest(HttpRequestInfo requestInfo)
   --- End of inner exception stack trace ---
   at OpenQA.Selenium.Appium.Service.AppiumCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Appium.AppiumDriver`1.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteWebDriver.FindElements(String mechanism, String value)
   at OpenQA.Selenium.Appium.AppiumDriver`1.FindElements(String selector, String value)
   at OpenQA.Selenium.Appium.AppiumDriver`1.FindElementsByAccessibilityId(String selector)
   at Xamarin.Forms.Core.UITests.WinDriverApp.QueryWindows(WinQuery query) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 823
   at Xamarin.Forms.Core.UITests.WinDriverApp.QueryWindows(Func`2 query) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 849
   at Xamarin.Forms.Core.UITests.WinDriverApp.<>c__DisplayClass91_0.<WaitForElement>b__0() in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 551
   at Xamarin.Forms.Core.UITests.WinDriverApp.Wait(Func`1 query, Func`2 satisfactory, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 989
   at Xamarin.Forms.Core.UITests.WinDriverApp.WaitForAtLeastOne(Func`1 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 1012
   at Xamarin.Forms.Core.UITests.WinDriverApp.WaitForElement(Func`2 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency, Nullable`1 postTimeout) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Core.Windows.UITests\WinDriverApp.cs:line 552
   at Xamarin.Forms.Controls.ScreenshotConditionalApp.WaitForElement(Func`2 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency, Nullable`1 postTimeout) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\TestPages\ScreenshotConditionalApp.cs:line 195
   at Xamarin.Forms.Controls.AppSetup.NavigateToIssue(Type type, IApp app) in D:\Workspaces\Xamarin.Forms\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\TestPages\TestPages.cs:line 182
[10/04/2019 10:38:18 Informational] NUnit VS Adapter 2.1.1.0 executing tests is finished
```

During the debug of the UITest I found out that the timeout occurs at the following line:
https://github.com/xamarin/Xamarin.Forms/blob/cbc9c442973ae5744ddcaed713e1910e3f293e82/Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/TestPages/TestPages.cs#L182

@samhouts Any idea what the issue could be? Problems with WinAppDriver (using V1.1)?

**Comment by samhouts** [comment]
closed by #5880
